components/DataGenerator.js

```diff
@@ -0,0 +1,220 @@
+              d +
+              chance.integer({ min: -10, max: 10 })
+          )
+        )
+      ),
+      toBeDeleted:
+        hour < 7
+          ? chance.bool({ likelihood: 20 })
+          : chance.bool({ likelihood: 5 }),
+
+      toBeDetailed: chance.bool({ likelihood: 20 }),
+    };
+
+    return object;
+  });
+
+  const daysEntries = hourlyEntries
+    //elimination of elements according to probability
+    .filter((entry) => entry.toBeDeleted !== true)
+    //adding isoDatestring for Filters and add details to some entries
+    .map((entry) => {
+      const isoDate = new Date(entry.timeStamp).toISOString();
+      entry.toBeDetailed === true
+        ? (entry = { ...generateDetailedEntry(), ...entry })
+        : entry;
+      const object = { ...entry, isoDate };
+
+      return object;
+    });
+
+  return daysEntries;
+}
+
+function generateDetailedEntry() {
+  const id = uid();
+  const randomEmotionNumber = chance.integer({ min: 0, max: 6 });
+  const randomStory = chance.integer({ min: 0, max: 2 });
+  const detailedEntry = {
+    id,
+    tensionLevel:
+      exampleData[randomEmotionNumber].stories[randomStory].tensionLevel,
+    emotion: exampleData[randomEmotionNumber].emotion,
+    intensity: chance.integer({ min: 40, max: 100 }),
+    category: exampleData[randomEmotionNumber].stories[randomStory].category,
+    trigger: exampleData[randomEmotionNumber].stories[randomStory].trigger,
+    notes: exampleData[randomEmotionNumber].stories[randomStory].notes,
+    isHighlighted: chance.bool({ likelihood: 90 }),
+  };
+
+  return detailedEntry;
+}
+
+function generateCompleteData(daysGoingBack) {
+  const currentFullDate = new Date();
+  currentFullDate.setHours(0);
+  const daysTimestamp = currentFullDate.valueOf();
+
+  const daysToFill = [...Array(Number(daysGoingBack)).keys()];
+  const dailyEntries = daysToFill.flatMap((day) => {
+    const timeStamp = daysTimestamp - (daysGoingBack - 1 - day) * 24 * 3600000;
+    const array = [...generateDaysEntries(23, timeStamp)];
+    return array;
+  });
+
+  function compare(a, b) {
+    if (a.timeStamp < b.timeStamp) {
+      return -1;
+    }
+    if (a.timeStamp > b.timeStamp) {
+      return 1;
+    }
+    return 0;
+  }
+
+  const completeSortedEntries = dailyEntries.sort(compare).reverse();
+  return completeSortedEntries;
+}
+
+export default function DataGenerator({
+  backupEntries,
+  setBackupEntries,
+  setShownEntries,
+  daysGoingBack,
+  setDaysGoingBack,
+  emotionEntries,
+  setEmotionEntries,
+}) {
+  return (
+    <>
+      <StyledGeneratorTitle>Data Generator</StyledGeneratorTitle>
+
+      <StyledFlexWrapper>
+        <p>Generate data for</p>
+        <label htmlFor="daysback">
+          <input
+            type="number"
+            id="daysGoingback"
+            value={daysGoingBack}
+            min={1}
+            max={150}
+            onChange={(event) => setDaysGoingBack(event.target.value)}
+          />
+          Days
+        </label>
```

Comment:
"Please ensure that the `htmlFor` attribute of the `<label>` tag matches the` id` attribute of the corresponding `<input>` tag.

```suggestion
        <label htmlFor="daysback">
          <input
            type="number"
            id="daysback"
            value={daysGoingBack}
            min={1}
            max={150}
            onChange={(event) => setDaysGoingBack(event.target.value)}
          />
          Days
        </label>
```

Reply:
Forgot to rename, is changed now.